sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -754,26 +755,38 @@ private[client] class Shim_v0_13 extends Shim_v0_12 {
           tryDirectSqlConfVar.defaultBoolVal.toString).toBoolean
         try {
           // Hive may throw an exception when calling this method in some circumstances, such as
-          // when filtering on a non-string partition column when the hive config key
-          // hive.metastore.try.direct.sql is false
+          // when filtering on a non-string partition column.
```

Comment:
To me your revised comment suggests that if we try to filter on a non-string partition column we can expect the call to `getPartitionedByFilter` to fail. That's not true.

Although I wrote the original comment, I can see how making an assumption about Hive's behavior when calling this method is rather specious and unwise. I suggest we just remove this comment entirely. To simply state that

```
Hive may throw an exception when calling this method in some circumstances.
```

only states the obvious—of course any method call may throw an exception in some circumstances.

Reply:
> of course any method call may throw an exception in some circumstances.
:+1: Yes, you are right. I agree with removing comments which might mislead future developers. I'll rephrase this.

Reply:
Done @mallman